elastic/eui
# [EuiSuperDatePicker] Convert date popover styles to Emotion

## Summary

Part 2 of 3 (see #7904, #7909)

This PR contains two conversions of class components to function components - I strongly recommend [following by commit](https://github.com/elastic/eui/pull/7908/commits) for easier review. The actual Sass conversions are fairly minimal, but getting `EuiAbsoluteTab` and `EuiRelativeTab` to a state where they can use hooks was the majority of the work.

## QA

https://eui.elastic.co/pr_7908/#/templates/super-date-picker should look the same as [production](https://eui.elastic.co/#/templates/super-date-picker) with the following permutations:

- Click the form control pretty date
- Click either of the two dates to trigger the date popover
- Absolute tab
  - [x] Padding should look the same as production
  - [x] Type invalid text (e.g. `asdf`) into the input. Confirm the formatting help text shows up and correctly overlaps into the submit button (should take up the full width of the popover)
  - [x] Press enter or click the submit button and confirm the help text goes red to indicate an error message
  - [ ] Select all text in the input and paste the following unix timestamp: `1698793930`. Confirm that the content automatically parses/validates on paste
- Relative tab
  - [x] Padding should look the same as production
  - [x] Confirm the round switch updates the date timestamp as expected
  - [x] Confirm the two selects update the date timestamp and parent super date picker control as expected
- Now tab
  - [x] Padding should look the same as production
- [x] Confirm the popover content (on all tabs) is the same width as production in desktop mode
- [x] Confirm the popover content sizes down to smaller width in mobile mode (~500px wide)

### General checklist

- Browser QA
    ~- [ ] Checked in both **light and dark** modes~ - Size CSS only, no colors
    ~- [ ] Checked for **accessibility** including keyboard-only and screenreader modes~
- Docs site QA - N/A
- Code quality checklist - N/A, no tests to update
- Release checklist
    ~- [ ] If applicable, added the **breaking change** issue label (and filled out the breaking change checklist)~
- Designer checklist - N/A
